cpp/betweenness_centrality_module/betweenness_centrality_test.cpp

```diff
@@ -0,0 +1,235 @@
+
+  std::vector<std::vector<uint64_t>> expected_predecessors({{}, {0}, {0}, {}, {}, {}, {}});
+  for (auto node_id = 0; node_id < number_of_nodes; node_id++) {
+    ASSERT_EQ(predecessors[node_id], expected_predecessors[node_id]);
+  }
+
+  std::vector<uint64_t> expected_shortest_paths_counter({1, 1, 1, 0, 0, 0, 0});
+  ASSERT_EQ(shortest_paths_counter, expected_shortest_paths_counter);
+}
+
+TEST(BetweennessCentralityUtilBFS, UndirectedCyclicGraph) {
+  //    (1)--(3)
+  //   / |    | \
+  // (0) |    |  (5)
+  //   \ |    | /
+  //    (2)--(4)
+  auto graph = mg_generate::BuildGraph(6, {{0, 1}, {0, 2}, {1, 3}, {1, 2}, {2, 4}, {3, 4}, {3, 5}, {4, 5}},
+                                       mg_graph::GraphType::kUndirectedGraph);
+  auto number_of_nodes = graph->Nodes().size();
+
+  // data structures used in BFS
+  std::stack<std::uint64_t> visited;
+  std::vector<std::vector<std::uint64_t>> predecessors(number_of_nodes, std::vector<std::uint64_t>());
+  std::vector<std::uint64_t> shortest_paths_counter(number_of_nodes, 0);
+  betweenness_centrality_util::BFS(0, *graph, visited, predecessors, shortest_paths_counter);
+
+  std::stack<uint64_t> expected_visited({0, 1, 2, 3, 4, 5});
+  ASSERT_EQ(visited, expected_visited);
+
+  std::vector<std::vector<uint64_t>> expected_predecessors({{}, {0}, {0}, {1}, {2}, {3, 4}});
+  for (auto node_id = 0; node_id < number_of_nodes; node_id++) {
+    ASSERT_EQ(predecessors[node_id], expected_predecessors[node_id]);
+  }
+
+  std::vector<uint64_t> expected_shortest_paths_counter({1, 1, 1, 1, 1, 2});
+  ASSERT_EQ(shortest_paths_counter, expected_shortest_paths_counter);
+}
+
+TEST(BetweennessCentralityUtilBFS, UndirectedMultipleShortestPaths) {
+  //    (1)        (5)
+  //   /   \      /   \
+  // (0)   (3)--(4)    (7)--(8)
+  //   \   /      \   /
+  //    (2)        (6)
+  auto graph =
+      mg_generate::BuildGraph(9, {{0, 1}, {0, 2}, {1, 3}, {2, 3}, {3, 4}, {4, 5}, {4, 6}, {5, 7}, {6, 7}, {7, 8}},
+                              mg_graph::GraphType::kUndirectedGraph);
+  auto number_of_nodes = graph->Nodes().size();
+
+  // data structures used in BFS
+  std::stack<std::uint64_t> visited;
+  std::vector<std::vector<std::uint64_t>> predecessors(number_of_nodes, std::vector<std::uint64_t>());
+  std::vector<std::uint64_t> shortest_paths_counter(number_of_nodes, 0);
+  betweenness_centrality_util::BFS(0, *graph, visited, predecessors, shortest_paths_counter);
+
+  std::stack<uint64_t> expected_visited({0, 1, 2, 3, 4, 5, 6, 7, 8});
+  ASSERT_EQ(visited, expected_visited);
+
+  std::vector<std::vector<uint64_t>> expected_predecessors({{}, {0}, {0}, {1, 2}, {3}, {4}, {4}, {5, 6}, {7}});
+  for (auto node_id = 0; node_id < number_of_nodes; node_id++) {
+    ASSERT_EQ(predecessors[node_id], expected_predecessors[node_id]);
+  }
+
+  std::vector<uint64_t> expected_shortest_paths_counter({1, 1, 1, 2, 2, 2, 2, 4, 4});
+  ASSERT_EQ(shortest_paths_counter, expected_shortest_paths_counter);
+}
```

Comment:
Maybe some quality of life changes, Gtest has parametrized tests (TEST_P) so you could send graph arguments (or created graph), expected_visited, expected_predecessors, and expected_shortest_paths_counter as parameters.

Reply:
@jmatak this could be something useful in general, because all the selected tests are the same with different data.

Reply:
Great idea, haven't really explored all the different functionalities of `gtest`.